qiskit_research/utils/pauli_twirling.py

```diff
@@ -109,19 +116,53 @@ def run(
     ) -> DAGCircuit:
         for run in dag.collect_runs(list(self.gates_to_twirl)):
             for node in run:
-                twirl_gates = TWIRL_GATES[node.op.name]
-                (before0, before1), (after0, after1) = twirl_gates[
-                    self.rng.integers(len(twirl_gates))
-                ]
-                mini_dag = DAGCircuit()
-                register = QuantumRegister(2)
-                mini_dag.add_qreg(register)
-                mini_dag.apply_operation_back(before0, [register[0]])
-                mini_dag.apply_operation_back(before1, [register[1]])
-                mini_dag.apply_operation_back(node.op, [register[0], register[1]])
-                mini_dag.apply_operation_back(after0, [register[0]])
-                mini_dag.apply_operation_back(after1, [register[1]])
-                dag.substitute_node_with_dag(node, mini_dag)
+                if len(node.op.params):
```

Comment:
The condition here doesn't seem quite right. I propose the following following if statement blocks to avoid having nested if statements (they should be in this order):
- The gate is in the `TWIRL_GATES` dictionary
- The gate is in the `TWO_QUBIT_ROTATION_GATES` list
- The gate is an `SECRGate`
- Else, throw an error

Reply:
Combined `SECRGate` into `TWO_QUBIT_PAULI_GENERATORS` (with `TWO_QUBIT_ROTATION_GATES`) and separated logic to treat `TWIRL_GATES` differently.